### PR TITLE
add unit tests for protected names

### LINKED ISSUE #265: add unit tests to assess protected names (ard_categorical and ard_continuous)

### PR DESCRIPTION
**Pull request by ayogasekaram**
**What changes are proposed in this pull request?**
- Added unit tests for protected names for `ard_continuous` and `ard_categorical`

closes #265 

--------------------------------------------------------------------------------

Pre-review Checklist (if item does not apply, mark is as complete)
- [ ] **All** GitHub Action workflows pass with a :white_check_mark:
- [ ] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [ ] If a bug was fixed, a unit test was added.
- [ ] Code coverage is suitable for any new functions/features (generally, 100% coverage for new code): `devtools::test_coverage()`
- [ ] Request a reviewer

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] If a bug was fixed, a unit test was added.
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`

When the branch is ready to be merged:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# cards (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] **All** GitHub Action workflows pass with a :white_check_mark:
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge" or "Rebase and merge".

### CODE REVIEW

**Comment by ayogasekaram** [comment]
I have read the CLA Document and I hereby sign the CLA

File: tests/testthat/_snaps/ard_categorical.md [hunk 1]
```diff
@@ -178,3 +178,11 @@
       Error in `ard_categorical()`:
       ! The `by` argument cannot include variables named "variable" and "variable_level".
 
+# ard_categorical.survey.design(by) messages about protected names
```
Comment (ddsjoberg) [inline tests/testthat/_snaps/ard_categorical.md:181]: can you update the title/test chunk name to not reference the survey methods?

**Review by ddsjoberg — CHANGES_REQUESTED** [review]
I think just one small change and we are good to go, thank you!!

**Review by ddsjoberg — APPROVED** [review]
Thank you @ayogasekaram !!

**Pull request merged**